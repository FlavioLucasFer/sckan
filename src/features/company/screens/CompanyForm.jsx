import React from 'react';
import PropTypes from 'prop-types';

import useFormInput from 'core/hooks/useFormInput';
import TextInput from 'core/components/TextInput';
import FormScreen from 'core/components/FormScreen';

CompanyForm.propTypes = {
	id: PropTypes.string,
}

function CompanyForm({ id }) {
	const companyName = useFormInput();
	const tradingName = useFormInput();
	const federalDocument = useFormInput();
	const email = useFormInput();
	
	return (
		<FormScreen title="Company"
			id={id}>
			<div className="row">
				<TextInput {...companyName}
					title="Company name"
					className="col s12 l6"
					type="text"
					icon="business"
					validate
					mandatory
					required />

				<TextInput {...tradingName} 
					title="Trading name"
					className="col s12 l6"
					type="text"
					icon="storefront"
					validate
					mandatory
					required />
			</div>

			<div className="row">
				<TextInput {...federalDocument}
					title="Federal document"
					className="col s12 l6"
					type="text"
					icon="assignment_ind"
					mask="federalDocument"
					validateBy="federalDocument"
					validate
					mandatory
					required />

				<TextInput {...email}
					title="Email name"
					className="col s12 l6"
					type="email"
					icon="email"
					validateBy="email"
					validate
					mandatory
					required />
			</div>
		</FormScreen>
	);
}

export default CompanyForm;