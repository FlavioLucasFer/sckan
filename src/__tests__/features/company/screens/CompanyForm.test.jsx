import React from 'react';
import { shallow } from 'enzyme';

import CompanyForm from 'features/company/screens/CompanyForm';

describe('Testing CompanyForm form component', () => {
	it('should render correctly', () => {
		const wrapper = shallow(
			<CompanyForm />
		);

		expect(wrapper.debug()).toMatchSnapshot();

		wrapper.setProps({
			id: '2446456',
		});

		expect(wrapper.debug()).toMatchSnapshot();
	});
});