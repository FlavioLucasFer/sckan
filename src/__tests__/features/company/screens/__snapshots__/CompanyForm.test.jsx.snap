// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CompanyForm form component should render correctly 1`] = `
"<FormScreen title=\\"Company\\" id={[undefined]}>
  <div className=\\"row\\">
    <TextInput value=\\"\\" onChange={[Function: handleChange]} title=\\"Company name\\" className=\\"col s12 l6\\" type=\\"text\\" icon=\\"business\\" validate={true} mandatory={true} required={true} />
    <TextInput value=\\"\\" onChange={[Function: handleChange]} title=\\"Trading name\\" className=\\"col s12 l6\\" type=\\"text\\" icon=\\"storefront\\" validate={true} mandatory={true} required={true} />
  </div>
  <div className=\\"row\\">
    <TextInput value=\\"\\" onChange={[Function: handleChange]} title=\\"Federal document\\" className=\\"col s12 l6\\" type=\\"text\\" icon=\\"assignment_ind\\" mask=\\"federalDocument\\" validateBy=\\"federalDocument\\" validate={true} mandatory={true} required={true} />
    <TextInput value=\\"\\" onChange={[Function: handleChange]} title=\\"Email name\\" className=\\"col s12 l6\\" type=\\"email\\" icon=\\"email\\" validateBy=\\"email\\" validate={true} mandatory={true} required={true} />
  </div>
</FormScreen>"
`;

exports[`Testing CompanyForm form component should render correctly 2`] = `
"<FormScreen title=\\"Company\\" id=\\"2446456\\">
  <div className=\\"row\\">
    <TextInput value=\\"\\" onChange={[Function: handleChange]} title=\\"Company name\\" className=\\"col s12 l6\\" type=\\"text\\" icon=\\"business\\" validate={true} mandatory={true} required={true} />
    <TextInput value=\\"\\" onChange={[Function: handleChange]} title=\\"Trading name\\" className=\\"col s12 l6\\" type=\\"text\\" icon=\\"storefront\\" validate={true} mandatory={true} required={true} />
  </div>
  <div className=\\"row\\">
    <TextInput value=\\"\\" onChange={[Function: handleChange]} title=\\"Federal document\\" className=\\"col s12 l6\\" type=\\"text\\" icon=\\"assignment_ind\\" mask=\\"federalDocument\\" validateBy=\\"federalDocument\\" validate={true} mandatory={true} required={true} />
    <TextInput value=\\"\\" onChange={[Function: handleChange]} title=\\"Email name\\" className=\\"col s12 l6\\" type=\\"email\\" icon=\\"email\\" validateBy=\\"email\\" validate={true} mandatory={true} required={true} />
  </div>
</FormScreen>"
`;
